src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/EmptyBlockCheck.java

```diff
@@ -148,6 +148,19 @@ public void visitToken(DetailAST ast) {
             leftCurly = slistToken;
         }
 
+        // special check for TokenTypes.LITERAL_CASE
```

Comment:
>Is commit better than the method with a good name?

You mean I should have used a new method instead ? And yes I think I should have done that !

As far as the other points are concerned, some logical errors arise due to my code which I couldn't foresee. I hope once they are resolved CIs would go green ! Also just to be clear what exactly do you mean by "UTs" ? unit tests, right !